### PR TITLE
GitX Support For Env CRUD operations , Import

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
- Added support for GitX in Environments CRUD apis. Added a new import API to import Environments from remote to Git.

## Tests Done:

- Validated all the test cases without GitX params.
- Validated the creation of GitX service at all scopes.
- Validated the updation of GitX service at all scopes.
- Validated the import of GitX service at all scopes.

## Some SS of Tests
INLINE ENV CREATION
Screenshot 2024-03-04 at 10 07 09 AM
Screenshot 2024-03-04 at 10 07 14 AM

REMOTE ENV CREATION
Screenshot 2024-03-04 at 10 14 56 AM
Screenshot 2024-03-04 at 10 14 44 AM

INLINE UPDATE
Screenshot 2024-03-04 at 10 17 06 AM

NO OPS WHEN NO RESOURCE ALREADY EXISTS
Screenshot 2024-03-04 at 10 15 50 AM

IMPORT REMOTE ENV:
Screenshot 2024-03-04 at 10 41 10 AM

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: harness/nextgen/README.md [hunk 1]
```diff
@@ -137,6 +137,8 @@ Class | Method | HTTP request | Description
 *EnvironmentsApi* | [**GetEnvironmentV2**](docs/EnvironmentsApi.md#getenvironmentv2) | **Get** /ng/api/environmentsV2/{environmentIdentifier} | Gets an Environment by identifier
 *EnvironmentsApi* | [**UpdateEnvironmentV2**](docs/EnvironmentsApi.md#updateenvironmentv2) | **Put** /ng/api/environmentsV2 | Update an Environment by identifier
 *EnvironmentsApi* | [**UpsertEnvironmentV2**](docs/EnvironmentsApi.md#upsertenvironmentv2) | **Put** /ng/api/environmentsV2/upsert | Upsert an Environment by identifier
+*EnvironmentsApi* | [**ImportEnvironment**](docs/EnvironmentsV2Api.md#importenvironment) | **Post** /ng/api/environmentsV2/import | Get Environment YAML from Git Repository
+*EnvironmentsApi* | [**MoveEnvironmentConfigs**](docs/EnvironmentsV2Api.md#moveenvironmentconfigs) | **Post** /ng/api/environmentsV2/move-config/{environmentIdentifier} | Move environment YAML from inline to remote
```
Comment (sourabh-awashti) [inline harness/nextgen/README.md:141]: remove this move config

File: harness/nextgen/api_environments.go [hunk 2]
```diff
@@ -32,12 +32,34 @@ EnvironmentsApiService Create an Environment
...
+     * @param "RepoName" (optional.String) - 
+     * @param "IsHarnessCodeRepo" (optional.Bool) - 
 @return ResponseDtoEnvironmentResponse
 */
 
 type EnvironmentsApiCreateEnvironmentV2Opts struct {
 	Body optional.Interface
+    Branch optional.String
+    RepoIdentifier optional.String
+    RootFolder optional.String
```
Comment (sourabh-awashti) [inline harness/nextgen/api_environments.go:54]: we can remvove this Repo identifier and Root folder, as we are not using it

File: harness/nextgen/api_environments.go [hunk 3]
```diff
@@ -773,16 +822,38 @@ EnvironmentsApiService Gets an Environment by identifier
  * @param environmentIdentifier Environment Identifier for the entity
  * @param accountIdentifier Account Identifier for the Entity.
  * @param optional nil or *EnvironmentsApiGetEnvironmentV2Opts - Optional Parameters:
-     * @param "OrgIdentifier" (optional.String) -  Organization Identifier for the Entity.
-     * @param "ProjectIdentifier" (optional.String) -  Project Identifier for the Entity.
-     * @param "Deleted" (optional.Bool) -  Specify whether Environment is deleted or not
+     * @param "OrgIdentifier" (optional.String) - 
+     * @param "ProjectIdentifier" (optional.String) - 
+     * @param "Deleted" (optional.Bool) - 
+     * @param "Branch" (optional.String) -
```
Comment (sourabh-awashti) [inline harness/nextgen/api_environments.go:828]: don't remove the desc of org , proj, deleted fields from here.

File: harness/nextgen/api_environments.go [hunk 4]
```diff
@@ -773,16 +822,38 @@ EnvironmentsApiService Gets an Environment by identifier
...
+    Deleted optional.Bool
+    Branch optional.String
+    RepoIdentifier optional.String
+    GetDefaultFromOtherRepo optional.Bool
+    ParentEntityConnectorRef optional.String
+    ParentEntityRepoName optional.String
+    ParentEntityAccountIdentifier optional.String
+    ParentEntityOrgIdentifier optional.String
+    ParentEntityProjectIdentifier optional.String
+    RepoName optional.String
```
Comment (sourabh-awashti) [inline harness/nextgen/api_environments.go:854]: can we cross check on whether we are using these fields which have parentEntity as their prefix?? if not then we can remove it

File: harness/nextgen/api_environments.go [hunk 5]
```diff
@@ -1067,14 +1168,40 @@ EnvironmentsApiService Update an Environment by identifier
...
 */
 
 type EnvironmentsApiUpdateEnvironmentV2Opts struct {
-	Body    optional.Interface
-	IfMatch optional.String
+	Body optional.Interface
+    IfMatch optional.String
+    Branch optional.String
+    RepoIdentifier optional.String
+    RootFolder optional.String
```
Comment (sourabh-awashti) [inline harness/nextgen/api_environments.go:1194]: please remove repoIdentifier and root folder.

File: harness/nextgen/api_environments.go [hunk 6]
```diff
@@ -1464,3 +1621,202 @@ func (a *EnvironmentsApiService) UpsertServiceOverride(ctx context.Context, acco
...
+    OrgIdentifier optional.String
+    ProjectIdentifier optional.String
+    ConnectorRef optional.String
+    RepoName optional.String
+    Branch optional.String
+    FilePath optional.String
+    CommitMsg optional.String
+    IsNewBranch optional.Bool
+    BaseBranch optional.String
+}
```
Comment (sourabh-awashti) [inline harness/nextgen/api_environments.go:1822]: please remove this unnecessary changes related to move config

File: harness/nextgen/docs/EnvironmentMoveConfigResponse.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# EnvironmentMoveConfigResponse
+
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/EnvironmentMoveConfigResponse.md:2]: we don't need this class

File: harness/nextgen/docs/EnvironmentsApi.md [hunk 8]
```diff
@@ -14,6 +14,7 @@ Method | HTTP request | Description
 [**UpdateEnvironmentV2**](EnvironmentsApi.md#UpdateEnvironmentV2) | **Put** /ng/api/environmentsV2 | Update an Environment by identifier
 [**UpsertEnvironmentV2**](EnvironmentsApi.md#UpsertEnvironmentV2) | **Put** /ng/api/environmentsV2/upsert | Upsert an Environment by identifier
 [**UpsertServiceOverride**](EnvironmentsApi.md#UpsertServiceOverride) | **Post** /ng/api/environmentsV2/serviceOverrides | Upsert
+[**ImportEnvironment**](EnvironmentsApi.md#ImportEnvironment) | **Post** ng/api/environmentV2/import | Get Environment YAML from Git Repository
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/EnvironmentsApi.md:17]: we have to update more information here, can you check service PR and see what all changes are to made

File: harness/nextgen/README.md (hunk 1 above)
Comment (senjucanon2) [reply harness/nextgen/README.md]: Done

File: harness/nextgen/api_environments.go (hunk 2 above)
Comment (senjucanon2) [reply harness/nextgen/api_environments.go]: Done

File: harness/nextgen/api_environments.go (hunk 6 above)
Comment (senjucanon2) [reply harness/nextgen/api_environments.go:1788]: Now it has the updated descriptions generated by the Swagger

File: harness/nextgen/docs/EnvironmentMoveConfigResponse.md (hunk 7 above)
Comment (senjucanon2) [reply harness/nextgen/docs/EnvironmentMoveConfigResponse.md]: Removed

File: harness/nextgen/docs/EnvironmentsApi.md [hunk 9]
```diff
@@ -391,3 +392,44 @@ Name | Type | Description  | Notes
...
+ **optional** | ***ServicesV2ApiImportServiceOpts** | optional parameters | nil if no parameters
+
+### Optional Parameters
+Optional parameters are passed through a pointer to a EnvironmentsV2ApiImportEnvironmentOpts struct
+Name | Type | Description  | Notes
+------------- | ------------- | ------------- | -------------
+
+ **orgIdentifier** | **optional.String**|  | 
+ **projectIdentifier** | **optional.String**|  | 
+ **serviceIdentifier** | **optional.String**|  |
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/EnvironmentsApi.md:414]: this should be envIdentifier

File: harness/nextgen/docs/EnvironmentsApi.md [hunk 10]
```diff
@@ -391,3 +392,44 @@ Name | Type | Description  | Notes
...
+> ResponseEnvironmentImportResponseDto ImportEnvironment(ctx, accountIdentifier, optional)
+Get Environment YAML from Git Repository
+
+### Required Parameters
+
+Name | Type | Description  | Notes
+------------- | ------------- | ------------- | -------------
+ **ctx** | **context.Context** | context for authentication, logging, cancellation, deadlines, tracing, etc.
+  **accountIdentifier** | **string**|  | 
+ **optional** | ***ServicesV2ApiImportServiceOpts** | optional parameters | nil if no parameters
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/EnvironmentsApi.md:405]: this should be envImportOpts

File: harness/nextgen/docs/EnvironmentsApi.md [hunk 11]
```diff
@@ -391,3 +392,44 @@ Name | Type | Description  | Notes
...
+ **connectorRef** | **optional.String**|  | 
+ **repoName** | **optional.String**|  | 
+ **branch** | **optional.String**|  | 
+ **filePath** | **optional.String**|  | 
+ **isForceImport** | **optional.Bool**|  | [default to false]
+ **isHarnessCodeRepo** | **optional.Bool**|  | 
+
+### Return type
+
+[**ResponseServiceImportResponseDto**](ResponseEnvironmentImportResponseDto.md)
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/EnvironmentsApi.md:424]: changes this as well

File: harness/nextgen/api_environments.go (hunk 3 above)
Comment (senjucanon2) [reply harness/nextgen/api_environments.go:826]: Done

File: harness/nextgen/api_environments.go (hunk 4 above)
Comment (senjucanon2) [reply harness/nextgen/api_environments.go:845]: Removed Extra Params

File: harness/nextgen/api_environments.go (hunk 6 above)
Comment (senjucanon2) [reply harness/nextgen/api_environments.go:1758]: Removed Move Configs Api

File: harness/nextgen/docs/EnvironmentsApi.md (hunk 11 above)
Comment (senjucanon2) [reply harness/nextgen/docs/EnvironmentsApi.md:424]: Done

**Review by sourabh-awashti — APPROVED** [review]

**Review by lovish1999 — APPROVED** [review]
